src/serial.rs

```diff
@@ -103,6 +110,130 @@ cfg_if! {
     }
 }
 
+/// Types for configuring a serial interface.
+pub mod config {
+    // TODO: To export or not to export. As baud is used by this module,
+    // shouldn't this be automagically available to users of this module?
+    pub use crate::time::rate::{Baud, Extensions};
+
+    // Reexport stop bit enum from PAC. In case there is a breaking change,
+    // provide a compatible enum from this HAL.
+    pub use crate::pac::usart1::cr2::STOP_A as StopBits;
+
+    /// Parity generation and checking.
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    #[derive(Clone, Copy, PartialEq)]
+    pub enum Parity {
+        /// No parity bit will be added/checked.
+        ParityNone,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// even number of bits set. The parity is included in [`WordLength`].
+        ParityEven,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// odd number of bits set. The parity bit is included in
+        /// [`WordLength`].
+        ParityOdd,
+    }
+
+    /// Word length for transmitting/receiving. This is the length of all data
+    /// bits plus the parity bit in case odd or even parity is selected.
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    #[derive(Clone, Copy, PartialEq)]
+    pub enum WordLength {
+        // TODO: There are STM32F3 devices which support 7 data bits according
+        // to the reference manual RM0316, section 29.4 USART Implementation.
```

Comment:
Could this be implemented as 

```rust
pub enum WordLength {
    #[cfg(any(feature = "stm32f3...", feature ="..."))]
    DataAndParityBits8
}
```

? 

I don't know, if this is feasiable or annoying to work with, because the compile time conditional would now be needed at many places.